segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/Ledgers.java

```diff
@@ -64,14 +69,20 @@ static LedgerHandle create(BookKeeper bookKeeper, BookKeeperConfig config) throw
      * @param ledgerId   The Id of the Ledger to open.
      * @param bookKeeper A references to the BookKeeper client to use.
      * @param config     Configuration to use.
-     * @return A LedgerHandle for the newly opened ledger.
+     * @return A ReadHandle for the newly opened ledger.
      * @throws DurableDataLogException If an exception occurred. The causing exception is wrapped inside it.
      */
-    static LedgerHandle openFence(long ledgerId, BookKeeper bookKeeper, BookKeeperConfig config) throws DurableDataLogException {
+    static ReadHandle openFence(long ledgerId, BookKeeper bookKeeper, BookKeeperConfig config) throws DurableDataLogException {
         try {
             return Exceptions.handleInterruptedCall(
-                    () -> bookKeeper.openLedger(ledgerId, LEDGER_DIGEST_TYPE, config.getBKPassword()));
-        } catch (BKException bkEx) {
+                    () -> FutureUtils.result(bookKeeper
+                            .newOpenLedgerOp()
+                    .withLedgerId(ledgerId)
+                    .withDigestType(LEDGER_DIGEST_TYPE)
+                    .withPassword(config.getBKPassword())
+                    .withRecovery(true)
+                    .execute()));
+        } catch (Exception bkEx) {
```

Comment:
Same question here about changing the `result` call to one that takes an exception handler.

Reply:
updated with BK_EXCEPTION_HANDLER